### PR TITLE
[SUP-258] Improved web request handling

### PR DESCRIPTION
**Pull request by mduncan26**
https://raygun.atlassian.net/browse/SUP-258

Changes:
- Added and integrated a new post service

### CODE REVIEW

File: Mindscape.Raygun4Unity/RaygunClient.cs [hunk 1]
```diff
@@ -138,56 +162,27 @@ internal RaygunMessage BuildMessage(Exception exception, IList<string> tags, IDi
...
-            RaygunClient.Log(string.Format("Error serializing raygun message: {0}", ex.Message));
-          }
-
-          if (message != null)
-          {
-            SendMessage(message);
-          }
-        }
-      }
+      return string.IsNullOrEmpty(_apiKey) ? false : true;
```
Comment (samuel-holt) [inline Mindscape.Raygun4Unity/RaygunClient.cs:167]: `return !string.IsNullOrEmpty(_apiKey);`

**Review by samuel-holt — APPROVED** [review]
:+1: Code looks good, there's some Unity-specific stuff in here I'm not too familiar with, but it looks okay from a brief study of their API docs

File: Mindscape.Raygun4Unity/RaygunClient.cs (hunk 1 above)
Comment (QuantumNightmare) [reply Mindscape.Raygun4Unity/RaygunClient.cs:167]: +1

File: Mindscape.Raygun4Unity/RaygunCrashReportingPostService.cs [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+    private static void LogResponseCode(long responseCode)
+    {
+      if (responseCode == 400)
+      {
+        RaygunClient.Log("API Response: Bad message - could not parse the provided JSON");
+      }
+      else if (responseCode == 403)
+      {
+        RaygunClient.Log("API Response: Invalid API Key - The value specified in the header X-ApiKey did not match with a user");
```
Comment (QuantumNightmare) [inline Mindscape.Raygun4Unity/RaygunCrashReportingPostService.cs:51]: Suggestion: "did not match with an application"

**Review by QuantumNightmare — APPROVED** [review]
Looks good to me

**Pull request merged**